### PR TITLE
:sparkles: TK-4290: Added support for log files and resource limit

### PR DESCRIPTION
**Pull request by pallav-trilio**

### CODE REVIEW

File: tools/preflight/preflight.sh [hunk 1]
```diff
@@ -17,9 +19,8 @@ CHECK='\xE2\x9C\x94'
 CROSS='\xE2\x9D\x8C'
 
 MIN_HELM_VERSION="3.0.0"
-MIN_K8S_VERSION="1.18.0"
+MIN_K8S_VERSION="1.17.0"
```
Comment (umeshkumhar) [inline tools/preflight/preflight.sh:22]: I think we are targeting 1.18+ only .. @madhurtrilio?

File: tools/preflight/preflight.sh [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 set -o pipefail
 
+TIMESTAMP=$(date +%F:%M:%S)
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:7]: instead of `:` use `-`. Use this one `date +%F_%H-%M-%S`

File: tools/preflight/preflight.sh [hunk 3]
```diff
@@ -619,25 +666,25 @@ EOF
...
-    kubectl delete volumesnapshot -n "${NAMESPACE}" "${res}" --force --grace-period=0 --timeout=5s &>/dev/null || true
-    kubectl patch volumesnapshot -n "${NAMESPACE}" "${res}" --type=json -p='[{"op": "remove", "path": "/metadata/finalizers"}]' &>/dev/null || true
+    kubectl delete volumesnapshot -n "${NAMESPACE}" "${res}" --force --grace-period=0 --timeout=5s &>> preflight-log-${TIMESTAMP}.out || true
+    kubectl patch volumesnapshot -n "${NAMESPACE}" "${res}" --type=json -p='[{"op": "remove", "path": "/metadata/finalizers"}]' &>> preflight-log-${TIMESTAMP}.out || true
   done
 
-  kubectl delete --force --grace-period=0 pod "${SOURCE_POD}" "${RESTORE_POD}" "${UNUSED_RESTORE_POD}" &>/dev/null || true
+  kubectl delete --force --grace-period=0 pod "${SOURCE_POD}" "${RESTORE_POD}" "${UNUSED_RESTORE_POD}" &>> preflight-log-${TIMESTAMP}.out || true
 
-  kubectl delete all -l preflight-run="${RANDOM_STRING}" --force --grace-period=0 &>/dev/null || true
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:638]: why label changed? we need to remove only this CI pipeline job's pods

File: tools/preflight/preflight.sh [hunk 4]
```diff
@@ -36,6 +37,11 @@ UNUSED_RESTORE_PVC="unused-restored-pvc-${RANDOM_STRING}"
 UNUSED_VOLUME_SNAP_SRC="unused-source-pvc-${RANDOM_STRING}"
 DNS_UTILS="dnsutils-${RANDOM_STRING}"
 
+echolog()
+(
+    echo "$@" | tee -a preflight-log-${TIMESTAMP}.out
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:42]: change file extention to `.log`

File: tools/preflight/preflight.sh [hunk 5]
```diff
@@ -123,12 +129,12 @@ check_kubectl_access() {
   if [[ -n ${KUBECONFIG_PATH} ]]; then
     export KUBECONFIG=${KUBECONFIG_PATH}
   fi
-  echo -e "${LIGHT_BLUE}Checking access to the Kubernetes context $(kubectl config current-context)...${NC}\n"
+  echolog -e "${LIGHT_BLUE}Checking access to the Kubernetes context $(kubectl config current-context)...${NC}\n"
 
   if [[ $(kubectl get ns default) ]]; then
-    echo -e "${GREEN} ${CHECK} Able to access the default Kubernetes namespace${NC}\n"
+    echolog -e "${GREEN} ${CHECK} Able to access the default Kubernetes namespace${NC}\n"
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:135]: Will it be printed on stdout as well? Expectation is, such checks should be printed.

File: tools/preflight/preflight.sh [hunk 6]
```diff
@@ -313,39 +326,52 @@ spec:
...
 
-  cat <<EOF | kubectl apply -f - &>/dev/null
+  cat <<EOF | kubectl apply -f - &>> preflight-log-${TIMESTAMP}.out
 kind: PersistentVolumeClaim
 apiVersion: v1
 metadata:
   name: ${SOURCE_PVC}
   labels:
     trilio: tvk-preflight
-    preflight-run: ${RANDOM_STRING}
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:348]: do not remove this label

File: tools/preflight/preflight.sh [hunk 7]
```diff
@@ -483,44 +523,44 @@ spec:
       readOnly: false
 EOF
 
-  kubectl wait --for=condition=ready --timeout=5m pod/"${RESTORE_POD}" &>/dev/null
+  kubectl wait --for=condition=ready --timeout=2m pod/"${RESTORE_POD}" &>> preflight-log-${TIMESTAMP}.out
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:526]: make it 3m.. need to choose max safe value here

File: tools/preflight/preflight.sh [hunk 8]
```diff
@@ -150,77 +156,77 @@ check_if_ocp() {
...
   # Abort successfully in case of OCP setup
   if [[ $(check_if_ocp) == "Y" ]]; then
-    echo -e "${GREEN} ${CHECK} Helm not needed for OCP clusters${NC}\n"
+    echolog -e "${GREEN} ${CHECK} Helm not needed for OCP clusters${NC}\n"
     return ${exit_status}
   fi
 
-  if ! command -v "helm" >/dev/null 2>&1; then
-    echo -e "${RED} ${CROSS} Unable to find helm${NC}\n"
+  if ! command -v "helm" &>> preflight-log-${TIMESTAMP}.out; then
```
Comment (msurudwad) [inline tools/preflight/preflight.sh:172]: create variable this log file name and reuse

File: tools/preflight/preflight.sh (hunk 4 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:42]: resolved

File: tools/preflight/preflight.sh (hunk 7 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:526]: resolved

File: tools/preflight/preflight.sh (hunk 8 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:172]: resolved

File: tools/preflight/preflight.sh (hunk 6 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:348]: resolved

File: tools/preflight/preflight.sh (hunk 2 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:7]: resolved

File: tools/preflight/preflight.sh (hunk 3 above)
Comment (pallav-trilio) [reply tools/preflight/preflight.sh:638]: resolved

**Review by msurudwad — APPROVED** [review]

**Pull request merged**